core/blockchain.go

```diff
@@ -208,7 +207,7 @@ type BlockChain struct {
 	vmConfig   vm.Config
 
 	badBlocks          *lru.Cache                     // Bad block cache
-	shouldPreserve     func(*types.Block) bool        // Function used to determine whether should preserve the given block.
+	shouldPreserve     func(*types.Header) bool       // Function used to determine whether should preserve the given block.
```

Comment:
The `Block` data type is a superset of `Header`. The methods as implemented (and, afaik, intended) only need information from the Header. So this leans things up a little bit, and the refactor was very smooth across all uses.